src/cargo/ops/cargo_package.rs

```diff
@@ -135,9 +137,46 @@ pub fn package(ws: &Workspace<'_>, opts: &PackageOpts<'_>) -> CargoResult<Option
         fs::rename(&src_path, &dst_path)
             .with_context(|| "failed to move temporary tarball into final location")?;
     }
+
     Ok(Some(dst))
 }
 
+pub fn package(
+    ws_root: &Workspace<'_>,
+    opts: &PackageOpts<'_>,
+) -> CargoResult<Vec<Option<FileLock>>> {
+    let pkgs = ws_root.members_with_features(
+        &opts.to_package.to_package_id_specs(ws_root)?,
+        &opts.cli_features,
+    )?;
+    let mut dsts = Vec::with_capacity(pkgs.len());
+
+    for (pkg, cli_features) in pkgs {
+        let ws = Workspace::ephemeral(
```

Comment:
Instead of creating an ephemeral workspace per-package, could this pass in the existing workspace and the `Package` itself perhaps that's being packaged?